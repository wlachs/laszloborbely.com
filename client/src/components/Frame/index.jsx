import * as React from 'react';
import PropTypes from 'prop-types';
import './index.css';
import Navbar from '../Navbar';
import Footer from '../Footer';

function Frame(props) {
  const { children } = props;
  return (
    <div className="container col-12 m-0 p-0 min-vh-100 py-4">
      <Navbar />
      {children}
      <Footer />
    </div>
  );
}

Frame.propTypes = {
  children: PropTypes.element,
};

Frame.defaultProps = {
  children: null,
};

export default Frame;
