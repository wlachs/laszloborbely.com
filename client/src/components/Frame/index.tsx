import {type ReactNode} from 'react';
import './index.css';
import Navbar from '../Navbar/index.jsx';

function Frame(props: {children?: ReactNode}) {
	const {children} = props;
	return (
		<div className='frame-container col-12 m-0 p-0 min-vh-100 py-4 d-flex flex-column'>
			<Navbar/>
			{children}
		</div>
	);
}

export default Frame;
