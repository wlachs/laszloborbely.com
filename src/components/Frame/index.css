@import '../global.css';

.container {
    font-family: "Roboto", sans-serif;
    font-weight: 400;
    font-size: 18px;
}

h1 {
    color: var(--off-color);
    font-weight: 500;
    text-align: center;
    margin: 80px 0 33px 0;
}

h2 {
    color: var(--secondary-color);
    font-weight: 300;
    text-align: center;
    margin-bottom: 13px;
}

@media (min-width: 1200px) {
    h1 {
        font-size: 3rem !important;
    }

    h2 {
        font-size: 1.5rem !important;
    }
}

@media (max-width: 1199px) {
    h1 {
        font-size: 2.5rem !important;
    }

    h2 {
        font-size: 1.25rem !important;
    }
}

@media (max-width: 991px) {
    h1 {
        font-size: 2rem !important;
    }

    h2 {
        font-size: 1.25rem !important;
    }
}
