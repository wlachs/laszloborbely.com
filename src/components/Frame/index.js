import * as React from 'react';
import './index.css';
import Navbar from '../Navbar';

export default function Frame(props) {
    return (
        <div className='container col-12 col-sm-12 m-0 m-sm-0 p-0'>
            <Navbar {...props} />
            {props.children}
        </div>
    );
};
