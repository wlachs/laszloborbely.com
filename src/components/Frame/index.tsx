import './index.css';

import { type ReactNode } from 'react';

import Navbar from '../Navbar/index.jsx';

function Frame(props: { children?: ReactNode }) {
	const { children } = props;
	const classes = [
		'frame-container',
		'm-0',
		'mx-auto',
		'p-0',
		'py-lg-4',
		'min-vh-100',
	].join(' ');
	return (
		<div className={classes}>
			<Navbar />
			{children}
		</div>
	);
}

export default Frame;
