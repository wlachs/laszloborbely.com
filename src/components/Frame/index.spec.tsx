import { cleanup, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { afterEach, describe, expect, it } from 'vitest';

import Frame from './index.tsx';

describe('Frame component', () => {
	afterEach(() => {
		cleanup();
	});

	it('should always render child elements', () => {
		const child = 'child';
		const { container } = render(
			<MemoryRouter>
				<Frame>{child}</Frame>,
			</MemoryRouter>,
		);
		expect(screen.getByText(child)).toBeDefined();
		expect(container).toMatchSnapshot();
	});
});
