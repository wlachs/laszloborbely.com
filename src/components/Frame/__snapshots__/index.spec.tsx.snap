// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Frame component > should always render child elements 1`] = `
<div>
  <div
    class="container mx-auto px-4 sm:max-w-[50rem] lg:px-0 lg:py-2"
  >
    <div
      class="flex items-center justify-between py-4"
    >
      <a
        class="link"
        href="/blog"
      >
        <img
          alt="László Borbély - Full-Stack Software Engineer"
          class="w-12"
          src="/src/assets/images/logo.svg"
        />
      </a>
      <div
        class="flex flex-row-reverse items-center gap-3"
      >
        <a
          class="link"
          href="/contact"
        >
          Contact
        </a>
        <a
          class="link"
          href="/blog"
        >
          Blog
        </a>
      </div>
    </div>
    <div
      class="separator w-100 mb-3"
    />
    child
  </div>
  ,
</div>
`;
